app/models/deploy.rb

```diff
@@ -121,6 +121,34 @@ def self.expired
     joins(:job).where(jobs: { status: 'pending'} ).where("jobs.created_at < ?", threshold)
   end
 
+  def csv_buddy
+    if not (BuddyCheck.enabled? && stage.production?)
+      "Buddy Not Required"
+    elsif buddy.nil? && pending?
+      "Pending Buddy"
+    elsif buddy.nil?
+      "No Buddy"
+    elsif (user.id == buddy.id)
+      "Buddy Bypassed"
+    else
+      buddy.name
+    end
+  end
+
+  def self.to_csv
+    per = 1000
+    max = Deploy.count.fdiv(per).ceil
+    CSV.generate do |csv|
+      csv << ["Deploy Number", "Project Name", "Deploy Sumary", "Deploy Updated", "Deploy Created", "Deployer Name", "Buddy Name", "Production Flag", Deploy.count]
+      for page in 1..max
+        @deploys = Deploy.all.page(page).per(per)
```

Comment:
Use the find_each batching to perform this cleaner: http://guides.rubyonrails.org/active_record_querying.html#retrieving-multiple-objects-in-batches